bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/SitemapResource.java

```diff
@@ -471,16 +472,20 @@ private WidgetDTO createWidgetBean(String sitemapName, Widget widget, boolean dr
         }
         if (widget instanceof Video) {
             Video videoWidget = (Video) widget;
-            String wId = itemUIRegistry.getWidgetId(widget);
             if (videoWidget.getEncoding() != null) {
                 bean.encoding = videoWidget.getEncoding();
             }
-            if (uri.getPort() < 0 || uri.getPort() == 80) {
-                bean.url = uri.getScheme() + "://" + uri.getHost() + "/proxy?sitemap=" + sitemapName
-                        + ".sitemap&widgetId=" + wId;
+            if (videoWidget.getEncoding() != null && videoWidget.getEncoding().toLowerCase().contains("hls")) {
+                bean.url = videoWidget.getUrl();
             } else {
-                bean.url = uri.getScheme() + "://" + uri.getHost() + ":" + uri.getPort() + "/proxy?sitemap="
-                        + sitemapName + ".sitemap&widgetId=" + wId;
+                String wId = itemUIRegistry.getWidgetId(widget);
+                if (uri.getPort() < 0 || uri.getPort() == 80) {
```

Comment:
matter of taste: can this be refactored to use a `StringBuilder`? like:

```
StringBuilder sb = new StringBuilder();
sb.append(uri.getScheme()).append("://").append(uri.getHost());
if (uri.getPort() < 0 || uri.getPort() == 80) {
    sb.append(":").append(uri.getPort());
}
sb.append("/proxy?sitemap=").append(sitemapName).append(".sitemap&widgetId=").append(wId);
bean.url = sb.toString();
```

Reply:
Yes and a method could ne created as the same code is used twice, once for image and once for video.